dbt/adapters/mariadb/connections.py

```diff
@@ -16,10 +16,10 @@
 
 @dataclass(init=False)
 class MariaDBCredentials(Credentials):
-    server: str
+    server: str = ""
     port: Optional[int] = None
-    database: Optional[str] = None
-    schema: str
+    database: str = ""
```

Comment:
Will be interesting if this works. I remember there being some difficulties getting database+schema to play nicely together since MySQL uses two-part identifiers (`database.table`) rather than three-part identifiers (`database.schema.table`).

Reply:
We could easily ignore the MyPy error rather than strictly complying with the type that `dbt-core` uses, but this seems to be working fine. At least unit and integration tests are passing.